artifacts.go

```diff
@@ -23,14 +22,26 @@ const artifactsRouterPath = "/epr/{packageName}/{packageName:[a-z0-9_]+}-{packag
 
 var errArtifactNotFound = errors.New("artifact not found")
 
-func artifactsHandler(logger *zap.Logger, indexer Indexer, cacheTime time.Duration) func(w http.ResponseWriter, r *http.Request) {
-	return artifactsHandlerWithProxyMode(logger, indexer, proxymode.NoProxy(logger), cacheTime)
-}
-
-func artifactsHandlerWithProxyMode(logger *zap.Logger, indexer Indexer, proxyMode *proxymode.ProxyMode, cacheTime time.Duration) func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Incuding this `handlerOptions` struct, I think that this `artifactsHandlerWithProxyMode` could be removed and just keep one `artifactsHandler`. WDYT ?

Reply:
Related to this comment https://github.com/elastic/package-registry/pull/1382#discussion_r2266856353

With the latest change in https://github.com/elastic/package-registry/commit/876c5602bd0226d4abf55b9222e85103a17e10d9 , each handler can specify its own options. With that change, it also is noted which parameters are mandatory and which ones are optional.